custom/enikshay/tasks.py

```diff
@@ -313,27 +257,27 @@ def count_doses_taken(dose_taken_by_date, start_date=None, end_date=None):
             raise EnikshayTaskException("Both of start_date and end_date should be specified or niether of them")
 
         if not start_date:
-            return len([dose_taken for dose_taken in dose_taken_by_date.values() if dose_taken])
+            return len([status for status in dose_status_by_date.values() if status.taken])
         else:
             return len([
-                is_taken
-                for date, is_taken in dose_taken_by_date.iteritems()
-                if start_date <= date <= end_date and is_taken
+                status
+                for date, status in dose_status_by_date.iteritems()
+                if start_date <= date <= end_date and status.taken
             ])
 
     @staticmethod
-    def count_doses_taken_by_source(doses_taken_by_date, start_date=None, end_date=None):
+    def count_doses_taken_by_source(dose_status_by_date, start_date=None, end_date=None):
         """Count all sources of adherence and return the count within the desired timeframe
 
         {'99DOTS': 1, 'MERM': 1, 'treatment_supervisor': 0, ... }
         """
         counts = defaultdict(int)
-        for date, source in doses_taken_by_date.iteritems():
-            if source in VALID_ADHERENCE_SOURCES:
+        for date, status in dose_status_by_date.iteritems():
+            if status.source in VALID_ADHERENCE_SOURCES:
```

Comment:
This should preserve the original behavior, which is that a source set to `False`  doesn't count towards this metric.